dll/win32/shell32/CShellLink.cpp

```diff
@@ -1888,9 +1897,10 @@ HRESULT STDMETHODCALLTYPE CShellLink::SetIconLocation(LPCWSTR pszIconPath, INT i
          */
         // FIXME: http://stackoverflow.com/questions/2976489/ishelllinkseticonlocation-translates-my-icon-path-into-program-files-which-i
         // if (PathFullyUnExpandEnvStringsW(pszIconPath, szUnExpIconPath, _countof(szUnExpIconPath)))
-        SHExpandEnvironmentStringsW(pszIconPath, szUnExpIconPath, _countof(szUnExpIconPath));
+        PathUnExpandEnvStringsW(pszIconPath, szUnExpIconPath, _countof(szUnExpIconPath));
         if (wcscmp(pszIconPath, szUnExpIconPath) != 0)
         {
+            pszIconPath = szUnExpIconPath;
```

Comment:
Please move this just before the `pInfo = (LPEXP_SZ_LINK)...` line below.

Reply:
OK.